code/firmware/rosco_m68k_firmware/main1.c

```diff
@@ -114,9 +122,27 @@ static void initialize_loader_efp() {
     *program_loader_ptr = (uint32_t)default_program_loader;
 }
 
+#ifdef WITH_KERNEL
+static void initialize_kernel(void) {
+    api_init();
+    pmm_init();
+    pmm_free(0x20000, 0x30000);
```

Comment:
Will giving this range to the kernel cause issues with software running? When programs copy themselves from 0x40000 down to 0x2000, they could overlap with this region.

Reply:
After further reflection ([I'd forgotten](https://discord.com/channels/698525682199822367/711035186170822727/1166979412856668241) about things that work the way `sdfat_menu` does) I've moved this responsibility into the program for the time being.

Reply:
Sounds good :+1: It is annoying to have to do it this way (and it not "just work"), but I think it would have to be a (theoretically) breaking change for `BIN`-loaded programs, with everything beyond 0x2000 being owned by the software.